karapace/schema_registry_apis.py

```diff
@@ -294,8 +360,15 @@ def send_delete_subject_message(self, subject, version):
         value = '{{"subject":"{}","version":{}}}'.format(subject, version)
         return self.send_kafka_message(key, value)
 
-    async def compatibility_check(self, content_type, *, subject, version, request):
+    async def compatibility_check(
+        self, content_type: str, *, subject: str, version: str, request: HTTPRequest, user: Optional[User] = None
+    ) -> None:
         """Check for schema compatibility"""
+
+        if self._auth:
+            if not self._auth.check_authorization(user, Operation.Write, f"Subject:{subject}"):
+                self.r(body={"message": "Forbidden"}, content_type=JSON_CONTENT_TYPE, status=HTTPStatus.FORBIDDEN)
```

Comment:
This stuff should really be a method. It's repeated in so many places. The extracted method would take in the `Operation` and the resource. I'd also create do this  `f"Subject:{subject}"` in a single place in the code. So the method would take in the resource type (subject, config) and the resource "specifier", `None` in case of config and the subject name in the case of subject.